### ISSUE TITLE
`recommended` shared config

### ISSUE THREAD

**Issue by benmosher** [benmosher/eslint-plugin-import#402]
Seems to be a common pattern amongst plugins.
 
Probably should be a combo of `errors` and `warnings`?

Also: see #416, probably want the following parser options, too:

```js
  parserOptions: {
    sourceType: 'module',
    ecmaVersion: 6,
    ecmaFeatures: { experimentalObjectRestSpread: true, jsx: true },
  },
```

@jfmengels this is probably another way to steer users towards static analysis. I have felt like the `errors` and `warnings` shared configs were not named clearly, anyway. Probably will keep them since it doesn't hurt, but merge them (or maybe just take `errors`) for the `recommended` version.

**Comment by benmosher** [comment]
@jfmengels would love your feedback on what I put together for this

**Comment by jfmengels** [comment]
I left some comments in the commit. I'd be happy to discuss it more if you want to. Either in a new issue or in the commit.
